plugins/chat/app/controllers/api/chat_channel_threads_controller.rb

```diff
@@ -2,20 +2,11 @@
 
 class Chat::Api::ChatChannelThreadsController < Chat::Api
   def show
-    params.require(:channel_id)
-    params.require(:thread_id)
-
-    raise Discourse::NotFound if !SiteSetting.enable_experimental_chat_threaded_discussions
-
-    thread =
-      ChatThread
-        .includes(:channel)
-        .includes(original_message_user: :user_status)
-        .includes(original_message: :chat_webhook_event)
-        .find_by!(id: params[:thread_id], channel_id: params[:channel_id])
-
-    guardian.ensure_can_preview_chat_channel!(thread.channel)
-
-    render_serialized(thread, ChatThreadSerializer, root: "thread")
+    params.permit(:thread_id, :channel_id)
```

Comment:
We can remove this actually, as we’re checking input parameters directly in the service through its contract.